build-system/setup/__init__.py

```diff
@@ -28,6 +28,27 @@
 
 
 
+"""
+==============================================================================
+Name: path_documents
+==============================================================================
+"""
+
+def path_documents ():
+   if platform.system () == 'Windows':
+      import ctypes.wintypes
+      CSIDL_PERSONAL = 5       # My Documents
+      SHGFP_TYPE_CURRENT = 0   # Current, not default value
+
+      buf = ctypes.create_unicode_buffer (ctypes.wintypes.MAX_PATH)
+      ctypes.windll.shell32.SHGetFolderPathW (0, CSIDL_PERSONAL, 0, SHGFP_TYPE_CURRENT, buf)
+      return buf.value
+
+   else:
+      return os.path.join (os.path.expanduser ('~'), 'Documents')
```

Comment:
I'm wondering if we should put a bit more context here? Is the Juce implementation standard, or is it something specific to Juce? Apart from that looks great.

Reply:
More context has been added in a comment for the Windows section